presenters/slice_plotter_presenter.py

```diff
@@ -19,74 +33,138 @@
+            self._slice_view.error_invalid_y_params()
+            return
+        elif status == INVALID_X_PARAMS:
+            self._slice_view.error_invalid_x_params()
+            return
+        elif status == INVALID_PARAMS:
+            self._slice_view.error_invalid_plot_parameters()
+            return
+
         intensity_start = self._slice_view.get_slice_intensity_start()
         intensity_end = self._slice_view.get_slice_intensity_end()
-        norm_to_one = self._slice_view.get_slice_is_norm_to_one()
+        norm_to_one = bool(self._slice_view.get_slice_is_norm_to_one())
         smoothing = self._slice_view.get_slice_smoothing()
         colourmap = self._slice_view.get_slice_colourmap()
-        status = self._slice_plotter.display_slice(selected_workspace,x_axis, y_axis, smoothing, intensity_start,
-                                                   intensity_end, norm_to_one, colourmap)
-        if status == INVALID_PARAMS:
-            self._slice_view.error_invalid_plot_parameters()
-
-        elif status == INVALID_X_PARAMS:
-            self._slice_view.error_invalid_x_params()
-
-        elif status == INVALID_Y_PARAMS:
-            self._slice_view.error_invalid_y_params()
-
-        elif status == INVALID_INTENSITY:
+        try:
+            intensity_start = self._to_float(intensity_start)
+            intensity_end = self._to_float(intensity_end)
+        except ValueError:
             self._slice_view.error_invalid_intensity_params()
+            return
 
-        elif status == INVALID_SMOOTHING:
+        if intensity_start is not None and intensity_end is not None and intensity_start > intensity_end:
+            self._slice_view.error_invalid_intensity_params()
+            return
+        try:
+            smoothing = self._to_int(smoothing)
+        except ValueError:
             self._slice_view.error_invalid_smoothing_params()
 
-        elif status == INVALID_X_UNITS:
-            self._slice_view.error_invalid_x_units()
-
-        elif status == INVALID_Y_UNITS:
-            self._slice_view.error_invalid_y_units()
+        try:
+            plot_data, boundaries, colormap, norm = self._slice_plotter.display_slice(selected_workspace, x_axis,
+                                                                                      y_axis,
+                                                                                      smoothing, intensity_start,
+                                                                                      intensity_end, norm_to_one,
+                                                                                      colourmap)
+        except NotImplementedError as e:
+            self._slice_view.error(e.message)
+            return
+        try:
+            self._plot_module.imshow(plot_data, extent=boundaries, cmap=colourmap, aspect='auto', norm=norm,
+                                interpolation='none')
+        except ValueError:
```

Comment:
Apparently matplotlib is throwing a value error
I thought the only case that would happen was if intensity_min> data_max
could you send me the file you are testing with?
@martyngigg